Compiler warnings when building latest numba-dpex from source

building numba-dpex main is resulting in these warnings. They seem to have been introduced by recent changes.

```bash
-- Build files have been written to: /home/diptorupd/Desktop/devel/numba-dpex/_skbuild/linux-x86_64-3.11/cmake-build
[3/5] Building C object numba_dpex/core/runtime/CMakeFiles/_dpexrt_python.dir/_dpexrt_python.c.o
/home/diptorupd/Desktop/devel/numba-dpex/numba_dpex/core/runtime/_dpexrt_python.c:1265:26: warning: incompatible pointer types assigning to 'PyObject *' (aka 'struct _object *') from 'struct PySyclQueueObject *' [-Wincompatible-pointer-types]
    queue_struct->parent = queue_obj;
                         ^ ~~~~~~~~~
/home/diptorupd/Desktop/devel/numba-dpex/numba_dpex/core/runtime/_dpexrt_python.c:1302:19: warning: incompatible pointer types assigning to 'PyObject *' (aka 'struct _object *') from 'struct PySyclQueueObject *' [-Wincompatible-pointer-types]
        queue_obj = SyclQueue_Make(queuestruct->queue_ref);
                  ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/diptorupd/Desktop/devel/numba-dpex/numba_dpex/core/runtime/_dpexrt_python.c:1402:19: warning: incompatible pointer types assigning to 'PyObject *' (aka 'struct _object *') from 'struct PySyclEventObject *' [-Wincompatible-pointer-types]
        event_obj = SyclEvent_Make(eventstruct->event_ref);
                  ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

--- a/numba_dpex/core/runtime/_dpexrt_python.c
+++ b/numba_dpex/core/runtime/_dpexrt_python.c
@@ -1262,7 +1262,7 @@ static int DPEXRT_sycl_queue_from_python(NRT_api_functions *nrt,
     Py_INCREF(queue_obj);
     queue_struct->meminfo =
         nrt->manage_memory(queue_obj, NRT_MemInfo_pyobject_dtor);
-    queue_struct->parent = queue_obj;
+    queue_struct->parent = (PyObject *)queue_obj;
     queue_struct->queue_ref = queue_ref;
 
     return 0;
@@ -1299,7 +1299,7 @@ static PyObject *DPEXRT_sycl_queue_to_python(NRT_api_functions *nrt,
     if (queue_obj == NULL) {
         // Make create copy of queue_ref so we don't need to manage nrt lifetime
         // from python object.
-        queue_obj = SyclQueue_Make(queuestruct->queue_ref);
+        queue_obj = (PyObject *)SyclQueue_Make(queuestruct->queue_ref);
     }
     else {
         // Unfortunately we can not optimize (nrt->release that triggers
@@ -1399,7 +1399,7 @@ static PyObject *DPEXRT_sycl_event_to_python(NRT_api_functions *nrt,
     if (event_obj == NULL) {
         // Make create copy of event_ref so we don't need to manage nrt lifetime
         // from python object.
-        event_obj = SyclEvent_Make(eventstruct->event_ref);
+        event_obj = (PyObject *)SyclEvent_Make(eventstruct->event_ref);
     }
     else {
         // Unfortunately we can not optimize (nrt->release that triggers

